### PR TITLE
Activity Log

### PR DESCRIPTION
**Pull request by agalin920**

### CODE REVIEW

File: cypress/e2e/reports/activity-log/activity-log.spec.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  });
+
+  describe("Resource Tab", () => {
+    it("Navigates to Resource Detail on Resource Item click", () => {
+      cy.waitOn("/v1/env/audits*", () => {
+        cy.visit(
+          "/reports/activity-log/resources?from=2022-07-14&to=2022-07-16"
+        );
+      });
+      cy.wait(1000);
```
Comment (shrunyan) [inline cypress/e2e/reports/activity-log/activity-log.spec.js:117]: why an additional `wait` after the `waitOn`?

File: cypress/e2e/reports/activity-log/activity-log.spec.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        "/reports/activity-log/resources/7-f28fd4d4a9-qtjb66"
+      );
+    });
+  });
+
+  describe("Skeletons", () => {
+    it("Displays All Skeletons while initial API is called", () => {
+      cy.intercept("/v1/env/audits*", (req) => {
+        req.continue(async (res) => {
+          await new Promise((resolve) => setTimeout(resolve, 4000));
```
Comment (shrunyan) [inline cypress/e2e/reports/activity-log/activity-log.spec.js:130]: This is subtle complex. Please add a comment explaining. My understanding is we are inserting an explicit 4 seconds of time to ensure skeletons are rendered to test for their presence.

File: src/apps/content-editor/src/app/views/ItemEdit/Content/Actions/Widgets/WidgetDraftHistory/WidgetDraftHistory.js [hunk 3]
```diff
@@ -53,21 +48,35 @@ export default connect((state, props) => {
...
+            </ul>
+            {/* TODO: Replace this link with Zesty MUI AppLink */}
+            <Link
+              color="#FF5D03"
+              underline="hover"
+              href="#"
+              onClick={(evt) => {
+                evt.preventDefault();
+                history.push(
+                  `/reports/activity-log/resources/${props.itemZUID}`
```
Comment (shrunyan) [inline src/apps/content-editor/src/app/views/ItemEdit/Content/Actions/Widgets/WidgetDraftHistory/WidgetDraftHistory.js:71]: I won't mark this as blocking to deploy but we need a better abstraction for internal app link routing than supplanting default a tag behavior.

File: src/apps/reports/src/app/views/ActivityLog/components/ActionTimelineItem/TimelineItem.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+          variant="caption"
+          component="div"
+          color="text.secondary"
+          sx={{
+            whiteSpace: "nowrap",
+            overflow: "hidden",
+            textOverflow: "ellipsis",
+          }}
+        >
+          {" "}
```
Comment (shrunyan) [inline src/apps/reports/src/app/views/ActivityLog/components/ActionTimelineItem/TimelineItem.js:152]: Is this spacing intentional?

File: src/shell/index.js [hunk 5]
```diff
@@ -9,6 +9,8 @@ import { Router } from "react-router-dom";
 import { ThemeProvider, createTheme } from "@mui/material/styles";
 import { theme } from "@zesty-io/material";
 
+import "chart.js/auto";
```
Comment (shrunyan) [inline src/shell/index.js:12]: Why does this happen in the shell?

File: src/utility/resolveResourceType.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export const resolveResourceType = (ZUID) => {
```
Comment (shrunyan) [inline src/utility/resolveResourceType.js:1]: This should probably exist in this dep: https://github.com/zesty-io/zuid-js

File: src/utility/instanceZUID.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+const instanceZUID = window.location.host.split(".")[0];
```
Comment (shrunyan) [inline src/utility/instanceZUID.js:1]: This happens on app boot and is added to the store. Can the store value be selected instead of adding this util?

**Review by shrunyan — CHANGES_REQUESTED** [review]
The `.DS_Store` files should not be part of the PR. Please check the .gitignore and ensure those OS files are excluded.

File: src/apps/content-editor/src/app/views/ItemEdit/Content/Actions/Widgets/WidgetDraftHistory/WidgetDraftHistory.js (hunk 3 above)
Comment (agalin920) [reply src/apps/content-editor/src/app/views/ItemEdit/Content/Actions/Widgets/WidgetDraftHistory/WidgetDraftHistory.js:71]: Agreed I believe its marked with a TODO to replace with the yet to be completed Zesty MUI AppLink

File: src/utility/instanceZUID.js (hunk 7 above)
Comment (agalin920) [reply src/utility/instanceZUID.js:1]: This was added in the previous RTK Query implementation and it is used to construct the RTK Query base url so I don't think I would be able to select it from the store

**Comment by shrunyan** [comment]
@zcolah it looks like we do not have hover states on the resource list view items. Are these accounted for in the design?

File: src/shell/index.js (hunk 5 above)
Comment (agalin920) [reply src/shell/index.js:12]: Ah this is required for v3 and added it here to verify all charts. Will move

File: cypress/e2e/reports/activity-log/activity-log.spec.js (hunk 1 above)
Comment (agalin920) [reply cypress/e2e/reports/activity-log/activity-log.spec.js:117]: Testing purposes. Will remove

**Comment by shrunyan** [comment]
@agalin920 functional feedback.

- view button on the resource > item view is not deep linking me to the item edit view. seems to link to /content regardless of log type
- when clicking into an item from the resource list view it seems the query parameters are added introducing a additional browser history state causing the back button to not take me back to the resource list view, unless pressed multiple times.

**Comment by agalin920** [comment]
@shrunyan 
"view button on the resource > item view is not deep linking me to the item edit view. seems to link to /content regardless of log type "

This is caused by newly added log items not having proper URLs in the API response. Need to log a bug for it and transition to FE URL construction  

"when clicking into an item from  the resource list view it seems the query parameters are added introducing a additional browser history state causing the back button to not take me back to the resource list view, unless pressed multiple times"

This should be fixed now

**Review by zcolah — APPROVED** [review]
**Great job Andres. Future fixes we will make are:** 

1. Identifying a manner to incorporate a scroll bar into our UI
2. Reviewing text updating below Activity Log
3. Finalizing name (b/w Activity Log and Audit Log)
4. Calling out items with Missing Meta Titles properly 
5. Handle edge case of created not coming up for old items (before 2020)

**Review by shrunyan — APPROVED** [review]

**Pull request merged**